tests/HttpClientInterception.Tests/HttpRequestInterceptionBuilderTests.cs

```diff
@@ -2248,6 +2248,63 @@ public static async Task Can_Deregister_Custom_Matcher()
             () => client.GetStringAsync("https://google.com/"));
     }
 
+    [Fact]
+    public static async Task Can_Do_Custom_Matching_Even_Number_Requests_Return_200_Odd_Number_Requests_Return_429()
```

Comment:
Should this be added to the Tests `Examples.cs`?

Reply:
I think we only need to show one/the best way to demonstrate a specific scenario in the examples. That file is more for the documentation that to actually test the library.